spacy/ml/staticvectors.py

```diff
@@ -43,7 +43,7 @@ def forward(
         [doc.vocab.vectors.find(keys=doc.to_array(key_attr)) for doc in docs]
     )
     try:
-        vectors_data = model.ops.gemm(model.ops.as_contig(V[rows]), W, trans2=True)
+        vectors_data = model.ops.gemm(model.ops.as_contig(V[rows]) * (rows>=0).T[:, None], W, trans2=True)
```

Comment:
I think then following will introduce less copying, and should allow us to merge this less obtrusively.

```suggestion
        vectors_data = model.ops.as_contig(V[rows])
        # If any rows have a negative index, set the values of those vectors to 0.
        # This is placed behind an attr flag, so that we can merge this change prior to
        # v3.1 if we want. We could set the attr flag in the init function, which would 
        # introduce the behavior on models that are trained, but not old models that
        # are loaded.
        if model.attrs.get("set_negative_indices_to_zero"):
            vectors_data[rows < 0] = 0
        vectors_data = model.ops.gemm(vectors_data, W, trans2=True, out=vectors_data)
```

The implementation takes the data and makes it contiguous, and then sets the values of the negative rows to 0, before doing the multiplication.

Reply:
Hmm, I think the flag is excessive and just complicates the transition to v3.1. Looking at it again, maybe it would be better to just call the whole thing with different behavior `spacy.StaticVectors.v2`?

Because of how new vectors are added, negative indices should never be used to index into the table, so I don't think it's a sensible option to make it sound like negative indices should be indices. I could see adding an option for a positive index as `UNK` or something like that, but I don't like the sound of this option.

Reply:
Hmm yeah okay, the flag is a dumb idea. A v2 version would be better.

Reply:
Okay, to clarify: for now, just call it v2 with the 0-ing behavior and then if we want other options like an `UNK` row we can add that in the future?

Reply:
Ugh, this is messier than I thought since the architectures use the `StaticVectors` class directly.